### ISSUE TITLE
Regression: LinearRegionItem span parameter broken

### ISSUE THREAD

**Issue by 2xB** [pyqtgraph/pyqtgraph#1488]
### Short description
The blue area inside `LinearRegionItem`s does no longer correctly apply `span` since PR #1391 ( tagging @jjuod ). Maybe a bounding box issue?

Happy new year btw.!

### Code to reproduce

Minimal version of `Plotting.py`:
```python
import numpy as np
import pyqtgraph as pg

app = pg.mkQApp()

x2 = np.linspace(-100, 100, 1000)
data2 = np.sin(x2) / x2
p8 = pg.plot(data2, pen=(255,255,255,200))

lr = pg.LinearRegionItem([400,700], span=(0.5,1))
p8.addItem(lr)

app.exec_()
```

### Expected behavior
Blue region should be in top half of the view, also according to the documentation of `span`: https://pyqtgraph.readthedocs.io/en/latest/graphicsItems/linearregionitem.html?highlight=LinearRegionItem .

![image](https://user-images.githubusercontent.com/31772910/103490021-3f6a1d80-4e19-11eb-91b1-9372febe35c0.png)

This screenshot was done with the last working commit 2c02d58134b3b9ef55e762de6410ce3edadcd9c4 .

### Real behavior
Blue region vanishes since merge of #1391 in 39f9c6a6aa3327ffee8bbf353ac5248237bf2ebe .

![image](https://user-images.githubusercontent.com/31772910/103490028-53ae1a80-4e19-11eb-8caf-0d14a172fc7f.png)

### Tested environment(s)

 * PyQtGraph version: Current `master` branch ( e94f53ac20eca89465c0e6853e0bb28e158c3c46 ). First occuring with commit 39f9c6a6aa3327ffee8bbf353ac5248237bf2ebe .
 * Qt Python binding: PyQt5 5.15.1 Qt 5.15.1
 * Python version: Python 3.8.5
 * NumPy version: 1.19.2
 * Operating system: Linux 5.8.16-2-MANJARO x86_64 20.1.2 Mikah
 * Installation method: Executed in folder

**Comment by j9ac9k** [comment]
I keep thinking that this issue gets fixed.... welcome PRs to help address this. One attempt was made here:

https://github.com/pyqtgraph/pyqtgraph/pull/1452

**Comment by jjuod** [comment]
Ah, turns out some commands modified the rectangle retrieved from cache, so setting span to 0.5 halved the cached rectangle on every call... Other similar functions make a copy after retrieving the cache, so shouldn't be affected. Submitted PR #1491

**Comment by j9ac9k** [comment]
Merged #1491 Which should address this issue, thaks for reporting @2xB thanks for the PR @jjuod
